### PR TITLE
ENH: CA status codes

### PR DESCRIPTION
**Pull request by klauer**

### CODE REVIEW

File: caproto/_utils.py [hunk 1]
```diff
@@ -69,10 +70,6 @@ class RemoteProtocolError(ProtocolError):
     ...
 
 
-class CaprotoKeyError(CaprotoError, KeyError):
```
Comment (tacaswell) [inline caproto/_utils.py]: What you can not make up your mind on MRO order!

**Comment by tacaswell** [comment]
What do you do with these things now that we have them defined?

**Comment by klauer** [comment]
From the docs: 
> Return codes are communicated in the protocol by the CA_PROTO_READ_NOTIFY, CA_PROTO_WRITE_NOTIFY, monitor subscription responses, and the CA_PROTO_ERROR responses.

For such things as 'access denied' and 'bad channel ID' - we should be responding with appropriate errors instead of bailing out, eventually.

File: caproto/_utils.py (hunk 1 above)
Comment (klauer) [inline caproto/_utils.py]: _Seriously_, @danielballan!

**Comment by klauer** [comment]
Thanks, @danielballan. I preferred how this [other branch](https://github.com/klauer/caproto/commit/ff0df2f7964fa4dc4a8fa0142db98bbadef74779) looked, but couldn't quite get it into a nice final solution.

**Pull request merged**